File: java/콜라츠 수열 만들기.java
import java.util.ArrayList;

class Solution {
    public int[] solution(int n) {
        ArrayList<Integer> answer = new ArrayList<>();
        answer.add(n);
        while (n != 1) {
            if (n % 2 == 0) {
                n /= 2;
            }else {
                n = 3 * n + 1;
            }
            answer.add(n);
        }
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: javascript/콜라츠 수열 만들기.js
const solution = (n) => {
    let answer = [];
    answer.push(n);

    while(n !== 1) {
        if(n % 2 === 0) {
            n = n / 2;
        } else {
            n = 3 * n + 1;
        }
        answer.push(n);
    }
    return answer;
}